ext/date/php_date.c

```diff
@@ -3844,12 +3844,76 @@ PHP_METHOD(DateTimeImmutable, setTimestamp)
 }
 /* }}} */
 
+/* {{{ */
+PHP_METHOD(DateTimeImmutable, setMicroseconds)
```

Comment:
I know I'm late to the party, but the 2 methods should share the same implementation, shouldn't they? Only ~2-3 lines differ, which could be handled easily one way or another.

Reply:
What about existing code extending DateTime (like Carbon)?